src/ragger/utils/misc.py

```diff
@@ -47,10 +48,15 @@ def find_library_application(base_dir: Path, name: str, device: str) -> Path:
     return lib
 
 
-def find_main_application(base_dir: Path, build_dir: str, bin_name: str, device: str) -> Path:
+def find_main_application(base_dir: Path, device: str, sdk: str) -> Path:
     if not base_dir.is_dir():
         raise AssertionError(f"{base_dir} is not a directory")
-    app = Path(base_dir / build_dir / device / bin_name).resolve()
+    app = base_dir.resolve()
+    if sdk == "rust":
+        app_name = toml.load(base_dir / "Cargo.toml")["package"]["name"]
```

Comment:
Wouldn't it be better if the `ledgered` tool inferred this as well ? It seems like a better place than doing in it in Ragger.

Reply:
@agrojean-ledger Actually, it is already the case: `sdk` string is set from `manifest.app.sdk` passed as argument to `find_main_application` function.

Reply:
@agrojean-ledger Indeed there is quite some functions in this `utils` module which could be internalized into `ledgered`, including this one. Let's list that into [this issue](https://github.com/LedgerHQ/ledgered/issues/11) until I got time.